mne/datasets/sleep_physionet/tests/test_physionet.py

```diff
@@ -81,16 +81,48 @@ def test_run_update_age_records(tmpdir):
         pd.testing.assert_frame_equal(data, pd.read_csv(AGE_SLEEP_RECORDS))
 
 
+@pytest.mark.parametrize('subject', [39, 68, 69, 78, 79, 83])
+def test_sleep_physionet_age_missing_subjects(physionet_tmpdir, subject):
+    """Test handling of missing subjects in Sleep Physionet age fetcher."""
+    params = {'path': physionet_tmpdir, 'update_path': False}
+
+    with pytest.raises(
+            ValueError, match='This dataset contains subjects 0 to 82'):
+        age.fetch_data(
+            subjects=[subject], recording=[1], on_missing='raise', **params)
+    with pytest.warns(RuntimeWarning,
+                      match='This dataset contains subjects 0 to 82'):
+        age.fetch_data(
+            subjects=[subject], recording=[1], on_missing='warn', **params)
+    age.fetch_data(
+        subjects=[subject], recording=[1], on_missing='ignore', **params)
```

Comment:
if feel this test is going to take too long. Fetching 6 physionet subjects can take a while.

maybe use @pytest.mark.slowtest

@larsoner any better suggestion?

Reply:
I would use monkeypatch to make it actually not bother fetching the data. @hubertjb if this does not make sense I can push a quick commit

Reply:
Just in case - this test and the next one won't actually fetch the data, because the checks for valid subjects/recordings happen before any fetching happen, but most importantly, the subjects/recordings that are requested in the `@pytest.mark.parametrize` are not available anyway. Is the idea to use monkeypatching just in case a future change makes the tests actually fetch a file?

Reply:
I assumed (and @agramfort probably did, too) that they actually fetch data. If they don't then this is fine! Actually I would go a step further and `monkeypatch` our `utils._fetch_file` or so such that if it attempts to fetch a file in this part of the test it throws an error. Okay for me to push something like this?

Reply:
OK, good. That works for me @larsoner !

Reply:
if it does not fetch and it's fast then no need to do anything more complicated. I assumed reading the code that it would fetch something

Reply:
you can maybe just assert that number of files is empty if ignore